### ISSUE TITLE
AL0185 codeunit missing

### ISSUE THREAD

**Issue by peterz84** [HodorNV/ALOps#432]
I have been spending too much time on this, trying to figure out, why the compile step in my pipeline is failing. It must be something I'm overlooking:

I'm spinning up a 17.3 nl docker container, then I remove the BaseApp (because its replaced by a vendors baseapp. This seems to work. Then I'm publishing several apps, used in this database from the folder 'dependancies'.

Complete Yaml:
```
trigger: none
name: $(Build.BuildId)

variables:
- group: 'Secrets'  

pool: Business Central Pool

steps:
- checkout: self
  clean: true

- task: ALOpsDockerCreate@1
  displayName: 'ALOPS - Create Docker Image' 
  inputs:
    artifacttype: OnPrem
    artifactversion: '17.3'
    artifactcountry: 'nl'

- task: ALOpsDockerStart@1
  inputs:
    enable_symbol_loading: true
    memory_gb: '8'
    container_restart: 'always'
    docker_parameters: |
      --volume "C:\Licenses:C:/licenses"

- task: ALOpsDockerWait@1
  displayName: 'ALOPS - Wait for Docker Container to start'
  inputs:
    search_string: 'Ready for connections!'

- task: ALOpsLicenseImport@1
  displayName: 'ALOPS - License Import'
  inputs:
    usedocker: true
    license_path: 'C:\licenses\17.flf' 
    remove_license_file: false    

- task: ALOpsDockerExec@1
  displayName: 'ALOPS - Unpublish MS Base'
  inputs:
    scriptsource: 'Workfolder'
    script_location: '/devops/unpublishbase.ps1'

- task: ALOpsAppPublish@1
  displayName: 'ALOps Dependant extensions'
  inputs:
    usedocker: true
    skip_verification: true
    batch_publish_folder: '/.dependancies'

- task: ALOpsAppCompiler@1
  displayName: 'ALOPS - Compile Extension'
  inputs:
    usedocker: true
    nav_app_version: '1.0.0.*'
    targetproject: './app.json'   # point to the folder in your repo that you want to compile in this step
    failed_on_warnings: false

- task: ALOpsAppPublish@1
  displayName: 'Publish AL-App'
  inputs:
    usedocker: true
    nav_artifact_app_filter: '*.app'

- task: ALOpsDockerRemove@1
  displayName: 'ALOPS - Remove Docker Container'
  enabled: true
  condition: always()
```

Content of Unpublishbase.ps1:
```
$ServerInstance = "BC" 
$ApplicationName = "Base Application" 
$Publisher = "Microsoft" 

function UnpublishAppAndDependencies($ServerInstance, $ApplicationName) 
{ 
  Get-NAVAppInfo -ServerInstance $ServerInstance | Where-Object {  
  # If the dependencies of this extension include the application that we want to unpublish, it means we have to unpublish this application first. 
  (Get-NavAppInfo -ServerInstance $ServerInstance -Name $_.Name).Dependencies | Where-Object {$_.Name -eq $ApplicationName} 

 } | ForEach-Object { 
    UnpublishAppAndDependencies $ServerInstance $_.Name 
 } 
 Get-NAVAppInfo -ServerInstance $ServerInstance | where Name -eq $ApplicationName | Unpublish-NavApp -ServerInstance $ServerInstance -Name $ApplicationName 
} 

function UninstallAndUnpublish($ServerInstance, $ApplicationName) 
{ 
    Get-NAVAppInfo -ServerInstance $ServerInstance | where Name -eq $ApplicationName | Uninstall-NavApp -ServerInstance $ServerInstance -Name $ApplicationName -Force -Tenant default 
    UnpublishAppAndDependencies $ServerInstance  $ApplicationName 
} 

# Unpublish and uninstall the extensions  
Write-Host Unpublish and uninstall extensions 
UninstallAndUnpublish -ApplicationName $ApplicationName -ServerInstance $ServerInstance 
```

Content of folder dependancies:
![image](https://user-images.githubusercontent.com/25202313/135503858-f8ef5444-0a56-4eb4-9974-247ed330b780.png)

Log of Publish dependancies step:

```
2021-09-30T15:16:11.1255906Z ##[section]Starting: ALOps Dependant extensions
2021-09-30T15:16:11.1429777Z ==============================================================================
2021-09-30T15:16:11.1430140Z Task         : ALOps App Publish
2021-09-30T15:16:11.1430433Z Description  : Publish AL Extension to Business Central
2021-09-30T15:16:11.1430718Z Version      : 1.445.2623
2021-09-30T15:16:11.1430932Z Author       : Hodor
2021-09-30T15:16:11.1431227Z Help         : Publish Business Central extension to service tier.
2021-09-30T15:16:11.1431609Z ==============================================================================
2021-09-30T15:16:13.0178226Z *** Validate configuration
2021-09-30T15:16:13.4377233Z *** Task Inputs:
2021-09-30T15:16:13.4412141Z 
2021-09-30T15:16:13.4515404Z name                                                                                                              value
2021-09-30T15:16:13.4518791Z ----                                                                                                              -----
2021-09-30T15:16:13.4520079Z usedocker                                                                                                          True
2021-09-30T15:16:13.4521249Z fixed_tag                                                                                                              
2021-09-30T15:16:13.4522381Z installpublishedapps                                                                                               True
2021-09-30T15:16:13.4523270Z installaltesttool                                                                                                 False
2021-09-30T15:16:13.4524431Z install_al_app_names                                                                             Tests-TestLibraries...
2021-09-30T15:16:13.4525539Z strictappnames                                                                                                    False
2021-09-30T15:16:13.4526546Z nav_serverinstance                                                                                                BC140
2021-09-30T15:16:13.4528182Z artifact_path                                                                                                          
2021-09-30T15:16:13.4529309Z nav_artifact_app_filter                                                                                           *.app
2021-09-30T15:16:13.4530442Z skip_verification                                                                                                  True
2021-09-30T15:16:13.4531614Z forceinstall                                                                                                      False
2021-09-30T15:16:13.4532783Z forcesync                                                                                                         False
2021-09-30T15:16:13.4534025Z publish_scope                                                                                                    Global
2021-09-30T15:16:13.4535199Z tenant                                                                                                          default
2021-09-30T15:16:13.4536363Z batch_publish_folder                                                                                     /.dependancies
2021-09-30T15:16:13.4537382Z publisherazureactivedirectorytenantid                                                                                  
2021-09-30T15:16:13.4537942Z 
2021-09-30T15:16:13.4556716Z 
2021-09-30T15:16:13.4562736Z 
2021-09-30T15:16:13.4769847Z *** For documentation, please visit   : https://www.alops.be/documentation
2021-09-30T15:16:13.4772588Z 
2021-09-30T15:16:13.6708494Z *** ALOps License:
2021-09-30T15:16:13.6732043Z   * Licensed To: xxx (Per Run License)
2021-09-30T15:16:13.6734654Z 
2021-09-30T15:16:13.6834795Z *** Importing required PS-Functions
2021-09-30T15:16:14.0462892Z *** Initiate Docker Session
2021-09-30T15:16:14.2713831Z *** Set Docker Container ErrorActionPreference = Stop
2021-09-30T15:16:14.5502993Z *** Start Batch App-Publish.
2021-09-30T15:16:14.6204066Z *** Copy Artifacts: [C:\agent\_work\10\s\.dependancies] to [c:\Run\DevOps\Artifacts-7698708d2fe343998ea6176990fc0587]
2021-09-30T15:16:31.2595193Z *** Import NAV/BC Management DLL's
2021-09-30T15:16:31.2596836Z *** Loading assemblies for ServiceTier [BC].
2021-09-30T15:16:31.5533277Z *** Loading assemblies from: [C:\program files\microsoft dynamics nav\170\service]
2021-09-30T15:16:31.5542983Z *** Selected module: [C:\program files\microsoft dynamics nav\170\service\Microsoft.Dynamics.Nav.Apps.Management.dll]
2021-09-30T15:16:31.5747860Z *** Loading module: [C:\program files\microsoft dynamics nav\170\service\Microsoft.Dynamics.Nav.Apps.Management.dll]
2021-09-30T15:16:31.5922258Z *** Selected module: [C:\program files\microsoft dynamics nav\170\service\Microsoft.Dynamics.Nav.Management.dll]
2021-09-30T15:16:31.5939969Z *** Loading module: [C:\program files\microsoft dynamics nav\170\service\Microsoft.Dynamics.Nav.Management.dll]
2021-09-30T15:16:31.8902168Z *** Get Available Applications from [c:\Run\DevOps\Artifacts-7698708d2fe343998ea6176990fc0587].
2021-09-30T15:16:32.1439512Z *** Get [*.app] files recursive from [c:\Run\DevOps\Artifacts-7698708d2fe343998ea6176990fc0587]
2021-09-30T15:16:32.6112493Z *** Found [8] files.
2021-09-30T15:16:32.7833324Z *** Extract App detail from [C:\Run\DevOps\Artifacts-7698708d2fe343998ea6176990fc0587\Apportunix_Document Creator_1.24.0.0.app]
2021-09-30T15:16:33.0269750Z *** Extract App detail from [C:\Run\DevOps\Artifacts-7698708d2fe343998ea6176990fc0587\To-Increase_Application_17.3.53827.0.app]
2021-09-30T15:16:33.1165024Z *** Extract App detail from [C:\Run\DevOps\Artifacts-7698708d2fe343998ea6176990fc0587\To-Increase_IEM Advanced Application_17.3.53827.0.app]
2021-09-30T15:16:33.1937709Z *** Extract App detail from [C:\Run\DevOps\Artifacts-7698708d2fe343998ea6176990fc0587\To-Increase_IEM Application_17.3.53827.0.app]
2021-09-30T15:16:33.2025096Z *** Extract App detail from [C:\Run\DevOps\Artifacts-7698708d2fe343998ea6176990fc0587\To-Increase_IEM Extended Application_17.3.53827.0.app]
2021-09-30T15:16:33.2318714Z *** Extract App detail from [C:\Run\DevOps\Artifacts-7698708d2fe343998ea6176990fc0587\To-Increase_IEM Lifecycle_17.3.53827.0.app]
2021-09-30T15:16:33.2532101Z *** Extract App detail from [C:\Run\DevOps\Artifacts-7698708d2fe343998ea6176990fc0587\To-Increase__Exclude_TI-Common_17.3.53290.0.app]
2021-09-30T15:16:33.5045757Z *** Extract App detail from [C:\Run\DevOps\Artifacts-7698708d2fe343998ea6176990fc0587\To-Increase__Exclude_TI-Lifecycle Base_17.3.53300.0.app]
2021-09-30T15:16:33.5357859Z *** Installing Applications:
2021-09-30T15:16:33.5358696Z 
2021-09-30T15:16:33.5365000Z **************************************
2021-09-30T15:16:33.5368707Z  * App File = [C:\Run\DevOps\Artifacts-7698708d2fe343998ea6176990fc0587\To-Increase__Exclude_TI-Common_17.3.53290.0.app]
2021-09-30T15:16:33.5369501Z 
2021-09-30T15:16:33.5374515Z  * App.ID        = 3d24078f-27f6-4e56-b448-15adbb4f9858
2021-09-30T15:16:33.5377812Z  * App.Name      = _Exclude_TI-Common
2021-09-30T15:16:33.5380675Z  * App.Publisher = To-Increase
2021-09-30T15:16:33.5383212Z  * App.Version   = 17.3.53290.0
2021-09-30T15:16:33.6384363Z  => Publishing App '_Exclude_TI-Common v17.3.53290.0'
2021-09-30T15:16:34.4725473Z WARNING: The package has been built on a server with a different version than the current server version. Package: 
2021-09-30T15:16:34.4773029Z 17.0.20458.21072, Server: 17.0.20458.20517.
2021-09-30T15:16:34.6815743Z *** Get Tenants
2021-09-30T15:16:35.0121521Z *** Sync Tenant default
2021-09-30T15:16:35.1003409Z *** Sync App [_Exclude_TI-Common] on tenant [default]
2021-09-30T15:16:36.7100991Z *** Install App [_Exclude_TI-Common] on tenant [default]
2021-09-30T15:16:37.7301321Z 
2021-09-30T15:16:38.3618664Z **************************************
2021-09-30T15:16:38.4242916Z 
2021-09-30T15:16:38.5521960Z 
2021-09-30T15:16:38.8153999Z **************************************
2021-09-30T15:16:39.6279945Z  * App File = [C:\Run\DevOps\Artifacts-7698708d2fe343998ea6176990fc0587\To-Increase_IEM Application_17.3.53827.0.app]
2021-09-30T15:16:39.6286077Z 
2021-09-30T15:16:41.0475839Z  * App.ID        = e2ab31a7-03d3-4284-832b-6fe9d3dfb45a
2021-09-30T15:16:41.2051347Z  * App.Name      = IEM Application
2021-09-30T15:16:41.2546029Z  * App.Publisher = To-Increase
2021-09-30T15:16:41.3949777Z  * App.Version   = 17.3.53827.0
2021-09-30T15:16:41.7819285Z  => Publishing App 'IEM Application v17.3.53827.0'
2021-09-30T15:17:35.0592851Z *** Get Tenants
2021-09-30T15:17:38.8883275Z *** Sync Tenant default
2021-09-30T15:17:38.9027316Z *** Sync App [IEM Application] on tenant [default]
2021-09-30T15:18:34.6815522Z *** Install App [IEM Application] on tenant [default]
2021-09-30T15:19:06.7875137Z 
2021-09-30T15:19:06.7880566Z **************************************
2021-09-30T15:19:06.7890596Z 
2021-09-30T15:19:06.7897594Z 
2021-09-30T15:19:06.7909523Z **************************************
2021-09-30T15:19:06.7923255Z  * App File = [C:\Run\DevOps\Artifacts-7698708d2fe343998ea6176990fc0587\To-Increase_Application_17.3.53827.0.app]
2021-09-30T15:19:06.7932010Z 
2021-09-30T15:19:06.7944264Z  * App.ID        = 0ff41c32-e178-4e5d-8d62-1d817500ea60
2021-09-30T15:19:06.7967026Z  * App.Name      = Application
2021-09-30T15:19:06.7974153Z  * App.Publisher = To-Increase
2021-09-30T15:19:06.7977839Z  * App.Version   = 17.3.53827.0
2021-09-30T15:19:06.8370434Z  => Publishing App 'Application v17.3.53827.0'
2021-09-30T15:19:06.8429540Z *** Get Tenants
2021-09-30T15:19:06.8783211Z *** Sync Tenant default
2021-09-30T15:19:06.9147801Z *** Sync App [Application] on tenant [default]
2021-09-30T15:19:06.9970814Z *** Install App [Application] on tenant [default]
2021-09-30T15:19:07.1514069Z 
2021-09-30T15:19:07.1531937Z **************************************
2021-09-30T15:19:07.1532476Z 
2021-09-30T15:19:07.1545810Z 
2021-09-30T15:19:07.1548493Z **************************************
2021-09-30T15:19:07.1557526Z  * App File = [C:\Run\DevOps\Artifacts-7698708d2fe343998ea6176990fc0587\To-Increase_IEM Extended Application_17.3.53827.0.app]
2021-09-30T15:19:07.1563867Z 
2021-09-30T15:19:07.1584145Z  * App.ID        = 7a70ba3c-4650-42b0-8088-d57a6efc4f9d
2021-09-30T15:19:07.1600134Z  * App.Name      = IEM Extended Application
2021-09-30T15:19:07.1608620Z  * App.Publisher = To-Increase
2021-09-30T15:19:07.1611824Z  * App.Version   = 17.3.53827.0
2021-09-30T15:19:07.2007160Z  => Publishing App 'IEM Extended Application v17.3.53827.0'
2021-09-30T15:19:07.2195213Z *** Get Tenants
2021-09-30T15:19:07.2811921Z *** Sync Tenant default
2021-09-30T15:19:07.2974443Z *** Sync App [IEM Extended Application] on tenant [default]
2021-09-30T15:19:08.0320193Z *** Install App [IEM Extended Application] on tenant [default]
2021-09-30T15:19:08.2483776Z 
2021-09-30T15:19:08.2631130Z **************************************
2021-09-30T15:19:08.2632244Z 
2021-09-30T15:19:08.2636995Z 
2021-09-30T15:19:08.2783452Z **************************************
2021-09-30T15:19:08.2910461Z  * App File = [C:\Run\DevOps\Artifacts-7698708d2fe343998ea6176990fc0587\To-Increase__Exclude_TI-Lifecycle Base_17.3.53300.0.app]
2021-09-30T15:19:08.2914931Z 
2021-09-30T15:19:08.2993557Z  * App.ID        = 82cc5a2f-409e-4ab8-b603-15dcfd9c11d7
2021-09-30T15:19:08.3149996Z  * App.Name      = _Exclude_TI-Lifecycle Base
2021-09-30T15:19:08.3202934Z  * App.Publisher = To-Increase
2021-09-30T15:19:08.3210417Z  * App.Version   = 17.3.53300.0
2021-09-30T15:19:08.3621184Z  => Publishing App '_Exclude_TI-Lifecycle Base v17.3.53300.0'
2021-09-30T15:19:08.3733728Z WARNING: The package has been built on a server with a different version than the current server version. Package: 
2021-09-30T15:19:08.3736315Z 17.0.20458.21072, Server: 17.0.20458.20517.
2021-09-30T15:19:08.5332148Z *** Get Tenants
2021-09-30T15:19:08.5805737Z *** Sync Tenant default
2021-09-30T15:19:08.5978558Z *** Sync App [_Exclude_TI-Lifecycle Base] on tenant [default]
2021-09-30T15:19:09.0431284Z *** Install App [_Exclude_TI-Lifecycle Base] on tenant [default]
2021-09-30T15:19:09.3331200Z 
2021-09-30T15:19:09.3349735Z **************************************
2021-09-30T15:19:09.3350247Z 
2021-09-30T15:19:09.3360431Z 
2021-09-30T15:19:09.3376547Z **************************************
2021-09-30T15:19:09.3384840Z  * App File = [C:\Run\DevOps\Artifacts-7698708d2fe343998ea6176990fc0587\Apportunix_Document Creator_1.24.0.0.app]
2021-09-30T15:19:09.3385468Z 
2021-09-30T15:19:09.3398326Z  * App.ID        = e55908d7-0741-4ffd-ab42-10ad0fa3f028
2021-09-30T15:19:09.3406881Z  * App.Name      = Document Creator
2021-09-30T15:19:09.3423383Z  * App.Publisher = Apportunix
2021-09-30T15:19:09.3431390Z  * App.Version   = 1.24.0.0
2021-09-30T15:19:09.3830365Z  => Publishing App 'Document Creator v1.24.0.0'
2021-09-30T15:19:12.5951307Z *** Get Tenants
2021-09-30T15:19:12.6364612Z *** Sync Tenant default
2021-09-30T15:19:12.6536018Z *** Sync App [Document Creator] on tenant [default]
2021-09-30T15:19:12.7448216Z *** Install App [Document Creator] on tenant [default]
2021-09-30T15:19:12.8652080Z 
2021-09-30T15:19:12.8660651Z **************************************
2021-09-30T15:19:12.8671620Z 
2021-09-30T15:19:12.8683447Z 
2021-09-30T15:19:12.8686339Z **************************************
2021-09-30T15:19:12.8697698Z  * App File = [C:\Run\DevOps\Artifacts-7698708d2fe343998ea6176990fc0587\To-Increase_IEM Advanced Application_17.3.53827.0.app]
2021-09-30T15:19:12.8707762Z 
2021-09-30T15:19:12.8721377Z  * App.ID        = b11afcb2-4902-4356-9d88-3a25ed15a1c6
2021-09-30T15:19:12.8737387Z  * App.Name      = IEM Advanced Application
2021-09-30T15:19:12.8740716Z  * App.Publisher = To-Increase
2021-09-30T15:19:12.8748697Z  * App.Version   = 17.3.53827.0
2021-09-30T15:19:12.9145402Z  => Publishing App 'IEM Advanced Application v17.3.53827.0'
2021-09-30T15:19:12.9176328Z *** Get Tenants
2021-09-30T15:19:12.9645321Z *** Sync Tenant default
2021-09-30T15:19:12.9773782Z *** Sync App [IEM Advanced Application] on tenant [default]
2021-09-30T15:19:13.0648509Z *** Install App [IEM Advanced Application] on tenant [default]
2021-09-30T15:19:13.2339029Z 
2021-09-30T15:19:13.2341904Z **************************************
2021-09-30T15:19:13.2353816Z 
2021-09-30T15:19:13.2365663Z 
2021-09-30T15:19:13.2368325Z **************************************
2021-09-30T15:19:13.2377641Z  * App File = [C:\Run\DevOps\Artifacts-7698708d2fe343998ea6176990fc0587\To-Increase_IEM Lifecycle_17.3.53827.0.app]
2021-09-30T15:19:13.2387927Z 
2021-09-30T15:19:13.2439751Z  * App.ID        = c10f4c48-a23b-4c15-ab07-0f239582c546
2021-09-30T15:19:13.2456133Z  * App.Name      = IEM Lifecycle
2021-09-30T15:19:13.2463640Z  * App.Publisher = To-Increase
2021-09-30T15:19:13.2467245Z  * App.Version   = 17.3.53827.0
2021-09-30T15:19:13.2818957Z  => Publishing App 'IEM Lifecycle v17.3.53827.0'
2021-09-30T15:19:13.5020261Z *** Get Tenants
2021-09-30T15:19:13.5502665Z *** Sync Tenant default
2021-09-30T15:19:13.5680129Z *** Sync App [IEM Lifecycle] on tenant [default]
2021-09-30T15:19:14.3383421Z *** Install App [IEM Lifecycle] on tenant [default]
2021-09-30T15:19:15.4394814Z 
2021-09-30T15:19:15.4419115Z **************************************
2021-09-30T15:19:15.4419776Z 
2021-09-30T15:19:15.4431090Z *** Batch App-Publish finished.
2021-09-30T15:19:15.4439745Z *** AL-Extension Published to ALOps Repository
2021-09-30T15:19:15.4788920Z ##[section]Async Command Start: Add Build Tag
2021-09-30T15:19:15.4789684Z Build '27821' has following tags now: myimage-onprem-17.3.20469.20605-nl-10.0.17763.2183-ltsc2019, BC.Version.17.3.20469.20605, BC.Platform.17.0.20458.20517, App._Exclude_TI-Common.17.3.53290.0
2021-09-30T15:19:15.4790283Z ##[section]Async Command End: Add Build Tag
2021-09-30T15:19:15.4791113Z ##[section]Async Command Start: Add Build Tag
2021-09-30T15:19:15.4792031Z Build '27821' has following tags now: myimage-onprem-17.3.20469.20605-nl-10.0.17763.2183-ltsc2019, BC.Version.17.3.20469.20605, BC.Platform.17.0.20458.20517, App.IEM Application.17.3.53827.0, App._Exclude_TI-Common.17.3.53290.0
2021-09-30T15:19:15.4792776Z ##[section]Async Command End: Add Build Tag
2021-09-30T15:19:15.4793594Z ##[section]Async Command Start: Add Build Tag
2021-09-30T15:19:15.4794442Z Build '27821' has following tags now: myimage-onprem-17.3.20469.20605-nl-10.0.17763.2183-ltsc2019, BC.Version.17.3.20469.20605, BC.Platform.17.0.20458.20517, App.IEM Application.17.3.53827.0, App._Exclude_TI-Common.17.3.53290.0, App.Application.17.3.53827.0
2021-09-30T15:19:15.4795165Z ##[section]Async Command End: Add Build Tag
2021-09-30T15:19:15.4795977Z ##[section]Async Command Start: Add Build Tag
2021-09-30T15:19:15.4796904Z Build '27821' has following tags now: myimage-onprem-17.3.20469.20605-nl-10.0.17763.2183-ltsc2019, BC.Version.17.3.20469.20605, BC.Platform.17.0.20458.20517, App.IEM Application.17.3.53827.0, App._Exclude_TI-Common.17.3.53290.0, App.Application.17.3.53827.0, App.IEM Extended Application.17.3.53827.0
2021-09-30T15:19:15.4797705Z ##[section]Async Command End: Add Build Tag
2021-09-30T15:19:15.4798517Z ##[section]Async Command Start: Add Build Tag
2021-09-30T15:19:15.4799521Z Build '27821' has following tags now: myimage-onprem-17.3.20469.20605-nl-10.0.17763.2183-ltsc2019, BC.Version.17.3.20469.20605, BC.Platform.17.0.20458.20517, App.IEM Application.17.3.53827.0, App._Exclude_TI-Common.17.3.53290.0, App.Application.17.3.53827.0, App.IEM Extended Application.17.3.53827.0, App._Exclude_TI-Lifecycle Base.17.3.53300.0
2021-09-30T15:19:15.4800392Z ##[section]Async Command End: Add Build Tag
2021-09-30T15:19:15.4801197Z ##[section]Async Command Start: Add Build Tag
2021-09-30T15:19:15.4802265Z Build '27821' has following tags now: myimage-onprem-17.3.20469.20605-nl-10.0.17763.2183-ltsc2019, BC.Version.17.3.20469.20605, BC.Platform.17.0.20458.20517, App.Document Creator.1.24.0.0, App.IEM Application.17.3.53827.0, App._Exclude_TI-Common.17.3.53290.0, App.Application.17.3.53827.0, App.IEM Extended Application.17.3.53827.0, App._Exclude_TI-Lifecycle Base.17.3.53300.0
2021-09-30T15:19:15.4803210Z ##[section]Async Command End: Add Build Tag
2021-09-30T15:19:15.4804019Z ##[section]Async Command Start: Add Build Tag
2021-09-30T15:19:15.4805204Z Build '27821' has following tags now: myimage-onprem-17.3.20469.20605-nl-10.0.17763.2183-ltsc2019, BC.Version.17.3.20469.20605, BC.Platform.17.0.20458.20517, App.Document Creator.1.24.0.0, App.IEM Application.17.3.53827.0, App._Exclude_TI-Common.17.3.53290.0, App.Application.17.3.53827.0, App.IEM Extended Application.17.3.53827.0, App._Exclude_TI-Lifecycle Base.17.3.53300.0, App.IEM Advanced Application.17.3.53827.0
2021-09-30T15:19:15.4806230Z ##[section]Async Command End: Add Build Tag
2021-09-30T15:19:15.4807047Z ##[section]Async Command Start: Add Build Tag
2021-09-30T15:19:15.4808259Z Build '27821' has following tags now: myimage-onprem-17.3.20469.20605-nl-10.0.17763.2183-ltsc2019, BC.Version.17.3.20469.20605, BC.Platform.17.0.20458.20517, App.Document Creator.1.24.0.0, App.IEM Application.17.3.53827.0, App._Exclude_TI-Common.17.3.53290.0, App.Application.17.3.53827.0, App.IEM Extended Application.17.3.53827.0, App._Exclude_TI-Lifecycle Base.17.3.53300.0, App.IEM Lifecycle.17.3.53827.0, App.IEM Advanced Application.17.3.53827.0
2021-09-30T15:19:15.4809337Z ##[section]Async Command End: Add Build Tag
2021-09-30T15:19:15.4810303Z ##[section]Finishing: ALOps Dependant extensions
```

Several error from warnings:
**##[error]c:\Run\DevOps\src\pageextensions\DUTWCompanyInformation.PageExt.al(8,51): warning AL0604: Use of implicit 'with' will be removed in the future. Qualify with 'Rec.'. This warning will become an error in a future release.
##[error]c:\Run\DevOps\src\reports\DUTWPurchaseInvoice.Report.al(984,29): warning AL0547: The event 'OnAfterInitReport' should not expose global variables. This warning will become an error in a future release.**

(should not fail on these right? because I have set failed_on_warnings: false in the compiler step)

The real struggle are the errors like these:
**##[error]c:\Run\DevOps\src\reports\DUTWTransportOrder.Report.al(259,36): error AL0185: Codeunit 'wDCR_ReportFunctions' is missing
##[error]c:\Run\DevOps\src\reports\DUTWTransportOrder.Report.al(260,35): error AL0185: Codeunit 'wDCR_TranslationMgt' is missing**

These codeunits do exist in the Document Creator app which my app is dependant on. What can I do?

**Comment by waldo1001** [comment]
Can you start with the basics and check the resulting container by removing the step
```yaml
- task: ALOpsDockerRemove@1
  displayName: 'ALOPS - Remove Docker Container'
  enabled: true
  condition: always()
```
and after the pipeline, see what has been installed?  This way, you can figure out what actually failed and what not.

**Comment by peterz84** [comment]
Here it is:
![image](https://user-images.githubusercontent.com/25202313/135512301-65ada8d0-287d-4bc2-89ea-450f09eecffd.png)

**Comment by peterz84** [comment]
I even tried installing VScode on the agent machine, get the code from DevOps and download the symbols from this docker, but downloading the symbols fail for some reason. 
- changing the server address in launch.json to https gives me:
Error: The SSL connection could not be established, see inner exception.
The remote certificate is invalid according to the validation procedure
- changing the server address in launch.json to http gives me:
Unable to read data from the transport connection: An existing connection was forcibly closed by the remote host..
An existing connection was forcibly closed by the remote host.

**Comment by peterz84** [comment]
Got a little further, but still don't understand. I backupped the SQL database and restored it locally in an onPrem installation. Now I was able to run my code against this database. First I downloaded the symbols, and the weird thing is, the Document Creator app is only 2 kb in size, while the same app in my dependancies folder (which gets installed in the pipeline) is 18 kb.

.alpackages folder:
![image](https://user-images.githubusercontent.com/25202313/135586753-622ea2cb-70e6-4d70-b828-9294eb5d17fa.png)
.dependancies folder:
![image](https://user-images.githubusercontent.com/25202313/135586872-2fd69945-a27c-47bb-b04c-cdc4f5608cc6.png)
